### DISCUSSION TITLE
permission denied for function crypto_aead_det_decrypt

### DISCUSSION THREAD

**Discussion by m-wong-dev** [supabase/supabase#11397] [Questions]
I'm implementing [Transparent Column Encryption](https://supabase.com/blog/transparent-column-encryption-with-postgres) in my project.  Following the steps in the blog, I was able to see the decrypted data correctly in the Supabase UI itself, but when I'm trying to access the data with supabase-js on the frontend, it's giving me this error:
```
{
  error: {
    code: '42501',
    details: null,
    hint: null,
    message: 'permission denied for function crypto_aead_det_decrypt'
  },
  data: null,
  count: null,
  status: 401,
  statusText: 'Unauthorized'
}
```

I've tried to re-grant anon permissions to the `public` schema but that didn't work.  I thought that maybe `crypto_aead_det_decrypt` is a database function so I tried to grant that specific function permissions to anon as well but it said that that isn't a function.

Any ideas?

**Comment by james2doyle** [comment]
You need to grant the user the permissions to use the function:

`GRANT EXECUTE ON FUNCTION pgsodium.crypto_aead_det_decrypt (bytea, bytea, uuid, bytea)
TO authenticated;`
